controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -703,7 +709,8 @@ public static boolean isEpochTransitionConsistent(final EpochTransitionRecord ep
         // verify that epoch transition record is consistent with segment table
         if (latest.getCreationEpoch() == epochTransitionRecord.getNewEpoch()) { // if segment table is updated
             epochTransitionRecord.newSegmentsWithRange.entrySet().forEach(segmentWithRange -> {
-                Optional<SegmentRecord> segmentOpt = SegmentRecord.readRecord(segmentIndex, segmentTable, getSegmentNumber(segmentWithRange.getKey()));
+                Optional<SegmentRecord> segmentOpt = SegmentRecord.readRecord(segmentIndex, segmentTable,
+                        getSegmentNumber(segmentWithRange.getKey()) - startingSegmentNumber);
```

Comment:
we should not do such computations here, rather change the api of segmentRecord.readRecord to take startingSegment number and perform this computation in there.

Reply:
Now `SegmentIndexRecord` is the one take care of this kind of computations, as the starting segment number is stored in the first position of the segment index.